rsocket/ResumeManager.h

```diff
@@ -73,10 +75,43 @@ class ResumeManager {
   // remote side.
   virtual ResumePosition impliedPosition() const = 0;
 
-  // This gets called when a new stream is opened.
-  virtual void onStreamOpen(StreamId streamId, FrameType frameType) = 0;
-
   // This gets called when a stream is closed.
   virtual void onStreamClosed(StreamId streamId) = 0;
+
+  // Return the StreamIds of locally originating REQUEST_STREAMs
+  virtual std::unordered_set<StreamId> getRequesterRequestStreamIds() = 0;
+
+  // Get allowance for the given StreamId.  This is called for situations where
+  // the local side is a publisher (REQUEST_STREAM Responder and
+  // REQUEST_CHANNEL).  This should return the allowance which the local side
+  // has received and hasn't fulfilled yet.
+  virtual uint32_t getPublisherAllowance(StreamId) = 0;
+
+  // Get allowance for the given StreamId.  This is called for situations where
+  // the local side is a consumer (REQUEST_STREAM Requester and
+  // REQUEST_CHANNEL).  This should return the allowance which has been sent to
+  // the remote side and hasn't been fulfilled yet.
+  virtual uint32_t getConsumerAllowance(StreamId) = 0;
+
+  // Increment the publisher allowance to be preserved for the StreamId
+  virtual void incrPublisherAllowance(StreamId, uint32_t) = 0;
```

Comment:
publisher/consumer can be inferred from stream id right? We shouldn't need all these methods.